### DISCUSSION TITLE
Recording Analog Sound to an SD card

### DISCUSSION THREAD

**Discussion by Watimongo1245** [pschatzmann/arduino-audio-tools#293] [Q&A] [answered]
Hello,
I hope you are doing well. I am a third-year electrical and electronics engineering student trying to implement a sound recorder as a part of the project I am carrying out during my internship.

I want to inquire from you if it is possible to use the Arduino-audio-tools library to record sound from the MAX9814 analog microphone to a microSD card via ESP32. If so, would you kindly let me know the example code/library name for doing just that?

Your opinion is highly welcome, and I look forward to hearing from you. I have also attached a picture of my schematic.(Side note:The microphone analog pin is at GPIO34)
![Sound Recorder Using ESP32](https://user-images.githubusercontent.com/82986145/183381909-33cf5784-1b8a-4af6-8b87-7f4e01c39d1b.png)

**Comment by pschatzmann** [comment, marked as answer]
I have never used a MAX9814 and you need to double check the properties yourself: It needs to provide an audio output signal osciallating between 0 and 3.3V!
To get started I suggest that you have a look at this example: https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-adc-serial where I used a cjmcu-622 Microphone Sensor as input.

To support a file just replace the AudioSink (Serial) with a file. Please note the only additional logic which is needed is the opening of the file in the beginning and the closing when you end the processing.

If you want to encode the data e.g. to MP3 or WAV you can find the relevant information here: https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio

**Reply by davidlutta** [reply]
Hey @Watimongo1245 were you able to implement this ?

**Comment by natedawggggggg** [comment]
Hi, I have a very similar project where I am trying to implement this. I'm not too good with Arduino code and am struggling to work out how to combine the two examples you have given. I can run the first one to display the waveform from my microphone but don't understand how to encode that and write it to my SD card. I got [E] AudioCopy.h : 253 - write to target has failed! from running the encoding example code. This might be because my SD pins are SD_CS 5,SPI_MOSI 23, SPI_MISO 19 and SPI_SCK 18 and am not sure where SD.begin(SdSpiConfig(PIN_CS, DEDICATED_SPI, SD_SCK_MHZ(2))) is getting its values from.

**Comment by pschatzmann** [comment]
So, why don't you just use 
SD.begin(SdSpiConfig(SD_CS, DEDICATED_SPI, SD_SCK_MHZ(2)))  or SD.begin(SdSpiConfig(5, DEDICATED_SPI, SD_SCK_MHZ(2))) 
You could also check the value PIN_CS by using Serial.println(PIN_CS) or look it up in AudioConfig.h 

And since - as far as I can see - you use already use the default values you could even leave this away completely...

**Reply by natedawggggggg** [reply]
Yep, of course.

But how do I get my microphone signal from the stream-adc-serial example to instead encode it to the SD card. I've been struggling to find an example to do just that.

**Comment by pschatzmann** [comment]
To me it is difficult to see where your problem is: A file is a Stream like e.g. Serial, so you would use this as copy destination. Maybe you should learn first how to create and write to files first w/o the audio framework and you need to decide on which of the different libraries you want to use: SD, SDFAT, SDMMC.

Here is a example that writes and reads to a File using the SD library: https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/streams-audiokit-sd-audiokit/streams-audiokit-sd-audiokit.ino
